utils/jwtUtil.js

```diff
@@ -122,8 +122,32 @@ function createToken(tokenName, userName) {
   return createdToken;
 }
 
+/**
+ * Get token name from Token
+ * @param  {object}   req - request object
+ * @param  {Function} cb - callback function
+ * @returns {string} - Token name if token provided, else null.
+ */
+function getTokenNameFromToken(req) {
+  return new Promise((resolve, reject) => {
+    if (req && req.headers && req.headers.authorization) {
+      jwt.verify(req.headers.authorization, env.tokenSecret, {},
+      (err, decodedData) => {
+        if (err !== null || !decodedData) {
+          return reject(err);
+        }
+
+        return resolve(decodedData.tokenname);
+      });
+    } else {
+      resolve(null);
+    }
+  });
+} // getTokenNameFromToken
+
 module.exports = {
   verifyToken,
   createToken,
   getUsernameFromToken,
+  getTokenNameFromToken,
```

Comment:
I think we should just have a single function which returns the user name AND token name... I don't think we need to have separate ones.